code/modules/crafting/recipes.dm

```diff
@@ -66,6 +66,158 @@
 	category = CAT_WEAPONRY
 	subcategory = CAT_WEAPON
 
+/datum/crafting_recipe/advancedegun
+	name = "Advanced Energy Gun"
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	result = list(/obj/item/gun/energy/gun/nuclear)
+	reqs = list(/obj/item/gun/energy/gun = 1,
+				/obj/item/stack/cable_coil = 5,
+				/obj/item/weaponcrafting/gunkit/nuclear = 1)
+	time = 20 SECONDS
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/advancedegun/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/gun)
+
+/datum/crafting_recipe/tempgun
+	name = "Temperature Gun"
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	result = list(/obj/item/gun/energy/temperature)
+	reqs = list(/obj/item/gun/energy/gun = 1,
+				/obj/item/stack/cable_coil = 5,
+				/obj/item/weaponcrafting/gunkit/temperature = 1)
+	time = 20 SECONDS
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/tempgun/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/gun)
+
+/datum/crafting_recipe/decloner
+	name = "Biological Demolecularisor"
```

Comment:
It's a little confusing that we call it decloner in the kit description, but call it this here. I'm not sure what the gun itself gets called by code (since I've honestly never seen it in game) but it might be good to standardize it.

Reply:
The real name is biological demolecularisor, decloner is more short form. It was called decloner in protolathe so I left it as is(probably something to do with the protolathes UI but not sure)